### ISSUE TITLE
Magic Number doesn't ignore 0 or 1 by default

### ISSUE THREAD

**Issue by hawaiianchimp** [eslint/eslint#8052]
Magic Number is adapted from [Buddy.js](https://github.com/danielstjules/buddy.js) where it states that: 

> Furthermore, specific values can be ignored, such as 0 and 1, which are ignored by default.

1 and 0 should be ignored since they are not magic numbers.

**Tell us about your environment**

* **ESLint Version:** v3.15.0
* **Node Version:** v6.6.0
* **npm Version:** v3.10.3

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**

```
{
  "extends": "airbnb-base",
  "globals": {
    "$": true,
    "_": true,
    "Backbone": true,
    "define": true,
    "exports": true,
    "jQuery": true,
    "module": true,
    "moment": true,
    "require": true
  },
  "env": {
    "browser": true
  },
  "rules": {
    "import/no-unresolved": "off",
    "import/no-extraneous-dependencies": "off",
    "import/extensions": "off",
    "class-methods-use-this": "off",
    "max-len": ["error", {
      "code": 100,
      "ignorePattern": "^import .*",
      "ignoreUrls": true
    }],
    "new-cap": ["error", {"capIsNewExceptions": ["Deferred"]}],
    "no-new": "off",
    "no-param-reassign": ["error", { "props": false }],
    "no-underscore-dangle": "off",
    "no-magic-numbers": "error"
  }
}
```

**What did you do? Please include the actual source code causing the issue.**

```js

  static printNoInputMessage() {
    console.error('No filename inputs provided.');
    process.exit(1);
  }

```

**What did you expect to happen?**
0 and 1 should be ignored by default

**What actually happened? Please include the actual, raw output from ESLint.**
0 and 1 are registered as lint errors

**Comment by ilyavolodin** [comment]
This was a conscious decision by the team. Having default configuration for the rule, even if it's consists of widely used defaults, like 0 and 1 might (and did) confuse users.
Original discussion here: https://github.com/eslint/eslint/issues/4193

**Comment by platinumazure** [comment]
:-1: from me. Users can add this to their configuration easily. In addition, our documentation is correct and we should not change our default options to match someone else's documentation.

**Comment by hawaiianchimp** [comment]
Thanks @ilyavolodin for the explanation and link to the original discussion. 👍
